packages/x-date-pickers/src/ClockPicker/ClockPicker.tsx

```diff
@@ -258,8 +258,11 @@ export const ClockPicker = React.forwardRef(function ClockPicker<TDate extends u
 
   const now = useNow<TDate>();
   const utils = useUtils<TDate>();
-  const midnight = utils.setSeconds(utils.setMinutes(utils.setHours(now, 0), 0), 0);
-  const dateOrMidnight = date || midnight;
+
+  const dateOrMidnight = React.useMemo(
```

Comment:
Without this `useMemo`, the `useCallback` and `useMemo` below are useless when `date` is null